Lib/traceback.py

```diff
@@ -1532,7 +1532,11 @@ def _compute_suggestion_error(exc_value, tb, wrong_name):
         # has the wrong name as attribute
         if 'self' in frame.f_locals:
             self = frame.f_locals['self']
-            if hasattr(self, wrong_name):
+            try:
+                has_wrong_name = hasattr(self, wrong_name)
+            except BaseException:
```

Comment:
This can swallow things like KeyboardInterrupt and MemoryError. We shouldn't do that.

Reply:
Then `Exception` only?

Reply:
Would that catch SystemExit?

Reply:
No :)

```python
>>> SystemExit.__mro__
(<class 'SystemExit'>, <class 'BaseException'>, <class 'object'>)
```